internal/controller/postgrescluster/pgbackrest.go

```diff
@@ -1634,6 +1649,9 @@ func (r *Reconciler) reconcilePostgresClusterDataSource(ctx context.Context,
 		return errors.WithStack(err)
 	}
 
+	// TODO(snapshots): If pgdata is being sourced by a VolumeSnapshot then don't perform a typical restore job;
+	// we only want to replay the WAL.
```

Comment:
❓ This sounds like the behavior in `pgbackrest.DedicatedSnapshotVolumeRestoreCommand`. If so, should this `TODO` go away?

Reply:
The `DedicatedSnapshotVolumeRestoreCommand` does the delta restore into the dedicated volume and then the volume is snapshotted. Theoretically, because of the way we backup/restore/snapshot, when a clone is created using a snapshot, we should be able to just replay WAL to get the clone up to speed with the source cluster. However, as of this moment, we still do a delta restore after the clone pgdata volume has been sourced with a snapshot. This TODO is a reminder to implement the WAL replay for clones made from snapshots in the future.